### ISSUE TITLE
[deckhouse] delivery process improvements to get rid of the bottlenecks

### ISSUE THREAD

**Issue by EvgenySamoylov** [deckhouse/deckhouse#1683]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

Now we are struggling to deliver the new code promptly but looks like it's still the bottleneck of the entire process.
The main goal we are aiming to reach in the epic is to deliver the new features to the release channels as fast as possible.
In other words, we need to reduce the time gap between our decision to release code and its actual delivery.

### Proposed Solution

The first step of the issue would be to analyze the current release process and suggest room for improvements.

**Comment by EvgenySamoylov** [comment]
Closed in favor of #1941
